lib/customValidators.js

```diff
@@ -239,9 +196,74 @@ function isSrvrSecProperArray(schemaType, specialSecTypes, secObj, secName) {
   return true;
 }
 
+/**
+ * Validates if parameters specified in the channel have corresponding parameters object defined and if name does not contain url parameters
+ * 
+ * @private
+ * @param  {Object} parsedJSON parsed AsyncAPI document
+ * @param  {String} asyncapiYAMLorJSON AsyncAPI document in string
+ * @param  {String} initialFormat information of the document was oryginally JSON or YAML
+ * @returns {Boolean} true in case the document is valid, otherwise throws ParserError
+ */
+function validateChannels(parsedJSON, asyncapiYAMLorJSON, initialFormat) {
+  const chnls = parsedJSON.channels;
+  if (!chnls) return true;
+
+  const chnlsMap = new Map(Object.entries(chnls));
+  const notProvidedParams = new Map();
+  const invalidChannelName = new Map();
+
+  chnlsMap.forEach((val, key) => {
+    const variables = parseUrlVariables(key);
+    const notProvidedChannelParams = notProvidedParams.get(tilde(key));
+    const parameters = parseUrlParameters(key);
+    if (!variables && !parameters)
+      return;
+    if (variables) {
+      setNotProvidedParams(variables, val, key, notProvidedChannelParams);
+    }
+
+    if (parameters) {
+      invalidChannelName.set(key,
+        parameters);
+    }
+  });
+
+  if (notProvidedParams.size) {
+    throw new ParserError({
+      type: validationError,
+      title: 'Not all channel parameters are described with parameter object',
+      parsedJSON,
+      validationErrors: groupValidationErrors('channels', 'channel does not have a corresponding parameter object for', notProvidedParams, asyncapiYAMLorJSON, initialFormat)
+    });
+  }
+
+  if (invalidChannelName.size) {
+    throw new ParserError({
+      type: validationError,
+      title: 'Channel names with parameters exist',
+      parsedJSON,
+      validationErrors: groupValidationErrors('channels', 'channels contain invalid name with url parameters ', invalidChannelName, asyncapiYAMLorJSON, initialFormat)
+    });
+  }
+
+  return true;
+
+  function setNotProvidedParams(variables, val, key, notProvidedChannelParams) {
```

Comment:
I think you can move this function outside this function, as a separate private helper function, because it has the potential to be used in other validations, just don't forget about jsdoc.

Why? because I'm pretty sure you will need to use it in query params validation too, like here
```
    if (parameters) {
      invalidChannelName.set(key,
        parameters);
    }
```

this logic is here for handling use case where more that once the error of the same title appear during validation. In other words, I'm pretty sure the current implementation is not able to handle a case where you have 2 different channels containing query parameters. Please add such a test case and you will see what I mean.